distributed-calculator/react-calculator/server.js

```diff
@@ -14,41 +18,92 @@ const daprUrl = `http://localhost:${daprPort}/v1.0/invoke`;
 const stateStoreName = `statestore`;
 const stateUrl = `http://localhost:${daprPort}/v1.0/state/${stateStoreName}`;
 
+const debugMode = process.env.DEBUG_MODE || false; // Emit debug level logs for State management
+
 /**
 The following routes forward requests (using pipe) from our React client to our dapr-enabled services. Our Dapr sidecar lives on localhost:<daprPort>. We invoke other Dapr enabled services by calling /v1.0/invoke/<DAPR_ID>/method/<SERVICE'S_ROUTE>.
 */
 
 app.post('/calculate/add', async (req, res) => {
-  const addUrl = `${daprUrl}/addapp/method/add`;
-  req.pipe(request(addUrl)).pipe(res);
+  const appUrl = `${daprUrl}/addapp/method/add`;
+
+  // Dapr invoke add app
+  const appResponse = await axios.post(appUrl, req.body);
+
+  // Return expected string result to client
+  const result = String(appResponse.data);
+ 
+  res.send(result); 
 });
 
 app.post('/calculate/subtract', async (req, res) => {
-  const subtractUrl = `${daprUrl}/subtractapp/method/subtract`;
-  req.pipe(request(subtractUrl)).pipe(res);
+  const appUrl = `${daprUrl}/subtractapp/method/subtract`;
+  
+  // Dapr invoke subtract app
+  const appResponse = await axios.post(appUrl, req.body);
+
+  // Return expected string result to client
+  const result = String(appResponse.data);
+  res.send(result); 
 });
 
 app.post('/calculate/multiply', async (req, res) => {
-  const multiplyUrl = `${daprUrl}/multiplyapp/method/multiply`;
-  req.pipe(request(multiplyUrl)).pipe(res);
+  const appUrl = `${daprUrl}/multiplyapp/method/multiply`;
+
+  // Dapr invoke multiply app
+  const appResponse = await axios.post(appUrl, req.body);
+
+  // Return expected string result to client
+  const result = String(appResponse.data);
+
+  res.send(result);
 });
 
 app.post('/calculate/divide', async (req, res) => {
-  const divideUrl = `${daprUrl}/divideapp/method/divide`;
-  req.pipe(request(divideUrl)).pipe(res);
+  const appUrl = `${daprUrl}/divideapp/method/divide`;
+
+  // Dapr invoke divide app
+  const appResponse = await axios.post(appUrl, req.body);
+
+  // Return expected string result to client
+  const result = String(Number(appResponse.data));
```

Comment:
No need for the `String(Number())` thing, what are you trying to accomplish with it?

Reply:
Casting is all handled now per your suggestion by this:

  // Return expected string result to client
  return res.send(`${appResponse.data}`);